### PR TITLE
Added documentation for Named aggregation in groupby.agg (Issue #18220)

### LINKED ISSUE #18220: DOC: DataFrame.groupy.agg with a list of tuples
#### Code Sample, a copy-pastable example if possible

```python
>>> df = pd.DataFrame({'key' : ['a', 'a', 'b', 'b', 'a'],
                       'data' : np.random.randn(5)})

>>> df.groupby('key')['data'].agg([('foo', 'mean')])
          foo
key          
a   -0.944872
b    0.893212

```
#### Problem description
I know its deprecated to rename columns with a dictionary of dictionaries in a groupby but a list of tuples also renames columns.

This behavior is not found in the docs. I accidentally discovered this a while back and then saw it again in Python for Data Analysis.

### PR DESCRIPTION
**Pull request by Ememobong28**
closes #18220

### CODE REVIEW

File: doc/source/user_guide/groupby.rst [hunk 1]
```diff
@@ -722,16 +722,45 @@ accepts the special syntax in :meth:`.DataFrameGroupBy.agg` and :meth:`.SeriesGr
...
+        'kind': ['cat', 'dog', 'cat', 'dog'],
+        'height': [9.1, 6.0, 9.5, 34.0],
+        'weight': [7.9, 7.5, 9.9, 198.0]
+    })
+
+To demonstrate "named aggregation," let's group the DataFrame by the 'kind' column and apply different aggregations to the 'height' and 'weight' columns:
+
+.. ipython:: python
+
+   result = animals.groupby('kind').agg(
```
Comment (mroeschke) [inline doc/source/user_guide/groupby.rst:744]: Can you put this in the groupby docstring instead?

File: pandas/etc/profile.d/micromamba.sh [hunk 2]
```diff
@@ -0,0 +1,66 @@
+# Copyright (C) 2012 Anaconda, Inc
```
Comment (mroeschke) [inline pandas/etc/profile.d/micromamba.sh:1]: This shouldn't be included

File: doc/source/user_guide/groupby.html [hunk 3]
```diff
@@ -0,0 +1,2945 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (mroeschke) [inline doc/source/user_guide/groupby.html:1]: This shouldn't be included

**Review by mroeschke — CHANGES_REQUESTED** [review]

File: doc/source/user_guide/groupby.rst (hunk 1 above)
Comment (Ememobong28) [reply doc/source/user_guide/groupby.rst:744]: Okay

Comment (Ememobong28) [reply doc/source/user_guide/groupby.rst:744]: Hello, How do I put this in a groupby docstring

Comment (mroeschke) [reply doc/source/user_guide/groupby.rst:744]: The example would go here https://github.com/pandas-dev/pandas/blob/fc308235f0ca6242a209c1fe19f71884fb5d50cd/pandas/core/groupby/generic.py#L1339

**Comment by mroeschke** [comment]
Closing in favor of https://github.com/pandas-dev/pandas/pull/54563

**Pull request closed without merging**